staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -67,65 +71,106 @@ type kmsPluginProbe struct {
 	l            *sync.Mutex
 }
 
+type kmsv2PluginProbe struct {
+	name string
+	ttl  time.Duration
+	envelopekmsv2.Service
+	lastResponse *kmsPluginHealthzResponse
+	l            *sync.Mutex
+}
+
 func (h *kmsPluginProbe) toHealthzCheck(idx int) healthz.HealthChecker {
 	return healthz.NamedCheck(fmt.Sprintf("kms-provider-%d", idx), func(r *http.Request) error {
 		return h.Check()
 	})
 }
 
+func (p *kmsv2PluginProbe) toHealthzCheck(idx int) healthz.HealthChecker {
+	return healthz.NamedCheck(fmt.Sprintf("kmsv2-provider-%d", idx), func(r *http.Request) error {
+		return p.Check()
+	})
+}
```

Comment:
Is it neccesary to change the name of the healthcheck? I worry about people depending on the name for exclusion, etc... There health check configs will break if the version of the underlying KMS provider changes.

Reply:
Not required! I've changed it back to `kms-provider-*`. The name of the plugin configured in `EncryptionConfiguration` should be sufficient to know which one is failing.